pkg/controller/controller.go

```diff
@@ -863,11 +874,63 @@ func convertAndFilterCatalogToNamespacedTypes(namespace string, in *osb.CatalogR
 	return serviceClasses, servicePlans, nil
 }
 
+// GenerateEscapedName takes in an OSB ID and filters
+// it to fit K8S name restrictions. It escapes all
+// characters except for lowercase alphanumerics (excluding z
+// as we use that for the escape character), hyphens, and
+// periods, changing them into their UTF-8 hex
+// equivalents separated by 'z'. Hyphens that come after periods and
+// periods that come after hyphens or periods are also escaped.
+// If the resulting string is over the K8S limit of 63 characters,
+// it appends the md5 hash of the original OSB ID to the first 31 characters
+// (minus trailing periods) of the escaped name.
+func GenerateEscapedName(externalID string) string {
```

Comment:
Do we want to export this? None of the test packages look like they are the separate _test package type. I suppose someone might want to import this to pre-prep their service names?

Reply:
@jboyd01 wanted this to be unit tested, so I made it public

Reply:
It doesn't need to be public to be unit tested. Everything is in one package.

Reply:
Doesn't need to be, but if it's going to be unit tested directly instead of being subsumed in the testing of another feature, it should be public.

Reply:
I do not understand. Why? It is directly testable as a private function.